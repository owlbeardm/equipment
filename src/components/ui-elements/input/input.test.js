import React from 'react'
import { shallow } from 'enzyme'

import Input from './input'

const validMetaProps = { touched: true, pristine: false, valid: true, invalid: false }
const invalidMetaProps = { touched: true, pristine: false, valid: false, invalid: true, error: 'Bad input' }

const setup = (props) => {
  const defaultProps = { input: { name: 'the-input' }, type: 'text', label: 'the input', disabled: false }
  return shallow(<Input {...defaultProps} {...props} />)
}

describe('rendering testing', () => {
  let wrapper

  test('renders without error', () => {
    wrapper = setup({ meta: invalidMetaProps })
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('valid input data', () => {
    beforeEach(() => {
      wrapper = setup({ meta: validMetaProps })
    })

    test('contains valid class', () => {
      const validClassNode = wrapper.find('.has-success')
      expect(validClassNode.length).toBe(1)
    })

    test('contains success icon', () => {
      const iconNode = wrapper.find('.material-icons')
      expect(iconNode.text()).toEqual('done')
    })

    test('error message is hidden', () => {
      const errorMsg = wrapper.find('[data-test="error-message"]')
      expect(errorMsg.length).toBe(0)
    })
  })

  describe('invalid input data', () => {
    beforeEach(() => {
      wrapper = setup({ meta: invalidMetaProps })
    })
    test('contains invalid class', () => {
      const invalidClassNode = wrapper.find('.has-danger')
      expect(invalidClassNode.length).toBe(1)
    })

    test('contains error icon', () => {
      const iconNode = wrapper.find('.material-icons')
      expect(iconNode.text()).toEqual('clear')
    })

    test('shows error message', () => {
      const errorMsg = wrapper.find('[data-test="error-message"]')
      expect(errorMsg.text()).toEqual(invalidMetaProps.error)
    })
  })
})
