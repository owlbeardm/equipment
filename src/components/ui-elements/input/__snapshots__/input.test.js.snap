// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering testing renders without error 1`] = `"<div class=\\"form-group bmd-form-group  has-danger\\"><label class=\\"control-label bmd-label-static\\" for=\\"the-input\\">the input</label><input type=\\"text\\" name=\\"the-input\\" id=\\"the-input\\" class=\\"form-control\\"/><span class=\\"form-control-feedback\\"><i class=\\"material-icons \\">clear</i></span><label class=\\"error\\" for=\\"the-input\\" data-test=\\"error-message\\">Bad input</label></div>"`;
